### ISSUE TITLE
NIRSpec FS exposure fails in resample_spec

### ISSUE THREAD

**Issue by hbushouse** [spacetelescope/jwst#2059]
The data set jw95115001001_02102_00001_nrs1 fails in the `resample_spec` step with the traceback:
```
2018-05-21 16:03:17,003 - stpipe.Spec2Pipeline.photom - INFO - Step photom done
2018-05-21 16:03:17,056 - stpipe.Spec2Pipeline.resample_spec - INFO - Step resample_spec running with args (<MultiSlitModel from jw95115001001_02102_00001_nrs1_rate.fits>,).
2018-05-21 16:03:17,926 - stpipe.Spec2Pipeline.resample_spec - WARNING - /Users/bushouse/anaconda3/envs/myjwst/lib/python3.5/site-packages/jwst-0.9.4a0.dev36-py3.5-macosx-10.7-x86_64.egg/jwst/resample/resample_spec.py:169: RuntimeWarning: Mean of empty slice
  lookup_table = np.nanmean(grid_slit.wavelength, axis=0)

2018-05-21 16:03:18,863 - stpipe.Spec2Pipeline.resample_spec - INFO - Resampling slit S200A1 (31, 429)
2018-05-21 16:03:19,378 - stpipe.Spec2Pipeline.resample_spec - WARNING - /Users/bushouse/anaconda3/envs/myjwst/lib/python3.5/site-packages/scipy/interpolate/interpolate.py:2556: RuntimeWarning: invalid value encountered in greater
  if not np.all(np.diff(p) > 0.):

----------------------------------------------------------------------
ERROR RUNNING STEP 'Spec2Pipeline':
    The points in dimension 0 must be strictly ascending
Traceback (most recent call last):
  File "/Users/bushouse/anaconda3/envs/myjwst/lib/python3.5/site-packages/jwst-0.9.4a0.dev36-py3.5-macosx-10.7-x86_64.egg/jwst/stpipe/cmdline.py", line 297, in step_from_cmdline
    step.run(*positional)
  File "/Users/bushouse/anaconda3/envs/myjwst/lib/python3.5/site-packages/jwst-0.9.4a0.dev36-py3.5-macosx-10.7-x86_64.egg/jwst/stpipe/step.py", line 391, in run
    step_result = self.process(*args)
  File "/Users/bushouse/anaconda3/envs/myjwst/lib/python3.5/site-packages/jwst-0.9.4a0.dev36-py3.5-macosx-10.7-x86_64.egg/jwst/pipeline/calwebb_spec2.py", line 87, in process
    asn.filename
  File "/Users/bushouse/anaconda3/envs/myjwst/lib/python3.5/site-packages/jwst-0.9.4a0.dev36-py3.5-macosx-10.7-x86_64.egg/jwst/pipeline/calwebb_spec2.py", line 252, in process_exposure_product
    resamp = self.resample_spec(input)
  File "/Users/bushouse/anaconda3/envs/myjwst/lib/python3.5/site-packages/jwst-0.9.4a0.dev36-py3.5-macosx-10.7-x86_64.egg/jwst/stpipe/step.py", line 391, in run
    step_result = self.process(*args)
  File "/Users/bushouse/anaconda3/envs/myjwst/lib/python3.5/site-packages/jwst-0.9.4a0.dev36-py3.5-macosx-10.7-x86_64.egg/jwst/resample/resample_spec_step.py", line 58, in process
    resamp.do_drizzle()
  File "/Users/bushouse/anaconda3/envs/myjwst/lib/python3.5/site-packages/jwst-0.9.4a0.dev36-py3.5-macosx-10.7-x86_64.egg/jwst/resample/resample_spec.py", line 388, in do_drizzle
    pscale_ratio=self.pscale_ratio)
  File "/Users/bushouse/anaconda3/envs/myjwst/lib/python3.5/site-packages/jwst-0.9.4a0.dev36-py3.5-macosx-10.7-x86_64.egg/jwst/resample/gwcs_drizzle.py", line 220, in add_image
    fillval=self.fillval)
  File "/Users/bushouse/anaconda3/envs/myjwst/lib/python3.5/site-packages/jwst-0.9.4a0.dev36-py3.5-macosx-10.7-x86_64.egg/jwst/resample/gwcs_drizzle.py", line 442, in dodrizzle
    pixmap = resample_utils.calc_gwcs_pixmap(input_wcs, output_wcs, insci.shape)
  File "/Users/bushouse/anaconda3/envs/myjwst/lib/python3.5/site-packages/jwst-0.9.4a0.dev36-py3.5-macosx-10.7-x86_64.egg/jwst/resample/resample_utils.py", line 118, in calc_gwcs_pixmap
    pixmap = np.dstack(reproject(in_wcs, out_wcs)(grid[0], grid[1]))
  File "/Users/bushouse/anaconda3/envs/myjwst/lib/python3.5/site-packages/jwst-0.9.4a0.dev36-py3.5-macosx-10.7-x86_64.egg/jwst/resample/resample_utils.py", line 166, in _reproject
    det = backward_transform(*tuple(flat_sky))
  File "/Users/bushouse/anaconda3/envs/myjwst/lib/python3.5/site-packages/astropy/modeling/core.py", line 382, in __call__
    ('equivalencies', None)])
  File "/Users/bushouse/anaconda3/envs/myjwst/lib/python3.5/site-packages/astropy/modeling/core.py", line 363, in __call__
    return super(cls, self).__call__(*inputs, **kwargs)
  File "/Users/bushouse/anaconda3/envs/myjwst/lib/python3.5/site-packages/astropy/modeling/core.py", line 815, in __call__
    outputs = self.evaluate(*chain(inputs, parameters))
  File "/Users/bushouse/anaconda3/envs/myjwst/lib/python3.5/site-packages/astropy/modeling/core.py", line 2921, in evaluate
    return self.__class__.evaluate(*args)
  File "/Users/bushouse/anaconda3/envs/myjwst/lib/python3.5/site-packages/astropy/modeling/core.py", line 2362, in evaluate
    result = cls._evaluate(inputs, params)
  File "/Users/bushouse/anaconda3/envs/myjwst/lib/python3.5/site-packages/astropy/modeling/core.py", line 2194, in <lambda>
    return (lambda inputs, params: g[0](f[0](inputs, params), params),
  File "/Users/bushouse/anaconda3/envs/myjwst/lib/python3.5/site-packages/astropy/modeling/core.py", line 2194, in <lambda>
    return (lambda inputs, params: g[0](f[0](inputs, params), params),
  File "/Users/bushouse/anaconda3/envs/myjwst/lib/python3.5/site-packages/astropy/modeling/core.py", line 2206, in <lambda>
    g[0](inputs[f[1]:], params)),
  File "/Users/bushouse/anaconda3/envs/myjwst/lib/python3.5/site-packages/astropy/modeling/core.py", line 2802, in f
    return evaluate_wrapper(model, inputs, param_values)
  File "/Users/bushouse/anaconda3/envs/myjwst/lib/python3.5/site-packages/astropy/modeling/core.py", line 2794, in evaluate_wrapper
    outputs = model.evaluate(*inputs, *param_values)
  File "/Users/bushouse/anaconda3/envs/myjwst/lib/python3.5/site-packages/astropy/modeling/tabular.py", line 225, in evaluate
    fill_value=self.fill_value)
  File "/Users/bushouse/anaconda3/envs/myjwst/lib/python3.5/site-packages/scipy/interpolate/interpolate.py", line 2558, in interpn
    "ascending" % i)
ValueError: The points in dimension 0 must be strictly ascending
```
This failure occurs when using the latest dev version of the cal pipeline, after having rerun the level-1b product all the way through `calwebb_detector1` and using the `_rate` product as input to `calwebb_spec2`.

**Comment by hbushouse** [comment]
Test files can be found in /grp/jwst/ssb/bushouse/jwst_data/NIRSpec/FS/

**Comment by jdavies-st** [comment]
This dataset now fails at the `pathloss` step:
```
2018-07-16 15:07:21,961 - stpipe.Spec2Pipeline.srctype - INFO - Step srctype done
2018-07-16 15:07:22,032 - stpipe.Spec2Pipeline.pathloss - INFO - Step pathloss running with args (<MultiSlitModel from jw95115001001_02102_00001_nrs1_rate.fits>,).
2018-07-16 15:07:22,073 - stpipe.Spec2Pipeline.pathloss - INFO - Using PATHLOSS reference file /grp/crds/cache/references/jwst/jwst_nirspec_pathloss_0001.fits
2018-07-16 15:07:22,314 - stpipe.Spec2Pipeline.pathloss - INFO - NRS_FIXEDSLIT
2018-07-16 15:07:22,314 - stpipe.Spec2Pipeline.pathloss - INFO - S200A1
2018-07-16 15:07:22,314 - stpipe.Spec2Pipeline.pathloss - INFO - Using aperture S200A1
Traceback (most recent call last):
  File "/Users/jdavies/dev/jwst/jwst/pipeline/calwebb_spec2.py", line 90, in process
    asn.filename
  File "/Users/jdavies/dev/jwst/jwst/pipeline/calwebb_spec2.py", line 241, in process_exposure_product
    input = self.pathloss(input)
  File "/Users/jdavies/dev/jwst/jwst/stpipe/step.py", line 400, in run
    step_result = self.process(*args)
  File "/Users/jdavies/dev/jwst/jwst/pathloss/pathloss_step.py", line 59, in process
    result = path_loss.do_correction(input_model, pathloss_model)
  File "/Users/jdavies/dev/jwst/jwst/pathloss/path_loss.py", line 284, in do_correction
    pathloss_pointsource_vector)
  File "/Users/jdavies/dev/jwst/jwst/pathloss/path_loss.py", line 384, in interpolate_onto_grid
    denominator = (wavelength_vector[upper_indices]
IndexError: index 21 is out of bounds for axis 1 with size 21
----------------------------------------------------------------------
ERROR RUNNING STEP 'Spec2Pipeline':
    One or more products failed to process. Failing calibration.
Traceback (most recent call last):
  File "/Users/jdavies/dev/jwst/jwst/stpipe/cmdline.py", line 297, in step_from_cmdline
    step.run(*positional)
  File "/Users/jdavies/dev/jwst/jwst/stpipe/step.py", line 400, in run
    step_result = self.process(*args)
  File "/Users/jdavies/dev/jwst/jwst/pipeline/calwebb_spec2.py", line 102, in process
    'One or more products failed to process. Failing calibration.'
RuntimeError: One or more products failed to process. Failing calibration.

----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/jdavies/dev/jwst/scripts/strun", line 14, in <module>
    step = Step.from_cmdline(sys.argv[1:])
  File "/Users/jdavies/dev/jwst/jwst/stpipe/step.py", line 162, in from_cmdline
    return cmdline.step_from_cmdline(args)
  File "/Users/jdavies/dev/jwst/jwst/stpipe/cmdline.py", line 297, in step_from_cmdline
    step.run(*positional)
  File "/Users/jdavies/dev/jwst/jwst/stpipe/step.py", line 400, in run
    step_result = self.process(*args)
  File "/Users/jdavies/dev/jwst/jwst/pipeline/calwebb_spec2.py", line 102, in process
    'One or more products failed to process. Failing calibration.'
RuntimeError: One or more products failed to process. Failing calibration.
```

**Comment by hbushouse** [comment]
If you skip or otherwise workaround the new problem in ``pathloss`` does it now succeed in ``resample_spec``?

**Comment by stscirij** [comment]
The pathloss step failed because the reference file wavelength coverage wasn't sufficient.  This is being tracked in https://jira.stsci.edu/browse/JP-348 and https://github.com/spacetelescope/jwst/issues/2252

**Comment by jdavies-st** [comment]
Yes @hbushouse, if you skip `pathloss`, it runs through `resample` now without problem, producing expected _s2d output.

**Comment by hbushouse** [comment]
Original issue fixed by #2245. Secondary issue (pathloss) fixed by #2276.
